### PR TITLE
factorize maven version

### PR DESCRIPTION
**Pull request by ctamisier**
re-push, I thought you merged it so I removed it. sorry.

### CODE REVIEW

File: generators/server/templates/_pom.xml [hunk 1]
```diff
@@ -56,8 +56,7 @@
         <%_ } _%>
         <%_ if (databaseType === 'sql') { _%>
         <hibernate.version>5.2.4.Final</hibernate.version>
-        <%_ } _%>
-        <%_ if (databaseType === 'sql') { _%>
+        <hibernate-validator.version>5.3.3.Final</hibernate-validator.version>
```
Comment (pascalgrimaud) [inline generators/server/templates/_pom.xml]: Must be before hibernate.version

File: generators/server/templates/_pom.xml [hunk 2]
```diff
@@ -73,6 +72,9 @@
         <%_ if (searchEngine === 'elasticsearch') { _%>
         <jna.version>4.2.2</jna.version>
         <%_ } _%>
+        <%_ if (authenticationType === 'jwt') { _%>
+        <jjwt.version>0.7.0</jjwt.version>
```
Comment (pascalgrimaud) [inline generators/server/templates/_pom.xml]: Must be before jna.version

**Comment by jdubois** [comment]
And do we have the same properties for Gradle? Both builds should be similar.

I understand the use of variables, but I didn't put any there because the version number is only used once, so it doesn't help a lot.

**Comment by jdubois** [comment]
Sorry for my last comment: of course this will become interesting when we do #4724

**Comment by ctamisier** [comment]
I'm just used to do this to have like an overview of the versions used for the dependencies..
Ok for the order. (I made the changes)
For gradle, jjwt version is already a variable.
And for hibernate validator I realize that it is inherited.
For maven the defined version in the parent is 5.2.4.Final and we override it to 5.3.3.Final.
I guess it should be the same for gradle. do I apply the change ?

**Comment by ctamisier** [comment]
Or maybe better to not override it. Were there some reasons to update the version ?

**Comment by ctamisier** [comment]
ok in gradle there is hibernate_validator_version=5.3.3.Final but the variable is not used. I guess I used it and that's it ? or we keep the spring boot parent one ?

**Comment by jdubois** [comment]
Have a look at the commit which made the Hibernate Validator version override - I think there is a good a reason for it, we don't override a Spring Boot-provided variable by random

**Comment by ctamisier** [comment]
https://github.com/jhipster/generator-jhipster/commit/631445b3c2517369d63020b62f570a57a4b7ca13

no special indication here... I will try to find something related in the issues.

**Comment by jdubois** [comment]
OK but the author is @gunnarmorling and he's the spec lead for Bean Validation, so I'm sure he knows this better than anyone else! I would keep the version number.

**Comment by ctamisier** [comment]
ok for the version.
I don't if hibernate_validator_version is used when the dependency is declared without the variable: 
`compile "org.hibernate:hibernate-validator"`

I am wondering if this is required:
`compile "org.hibernate:hibernate-validator:${hibernate_validator_version}"`

I don't know gradle and neither if hibernate_validator_version is declared in a "parent" (my mind is maven-oriented..)

**Comment by ctamisier** [comment]
For maven in dependency:tree we have
`[INFO] +- org.springframework.boot:spring-boot-starter-web:jar:1.4.2.RELEASE:compile`
`[INFO] |  +- org.hibernate:hibernate-validator:jar:5.3.3.Final:compile`

so we don't even need to declare:
`<dependency>`
`<groupId>org.hibernate</groupId>`
`<artifactId>hibernate-validator</artifactId>`
`<version>${hibernate-validator.version}</version>`
`</dependency>`

We can just have
`<properties>`
`...`
`<hibernate-validator.version>5.3.3.Final</hibernate-validator.version>`
`...`
`</properties>`

**Comment by ctamisier** [comment]
For Gradle I tried and 
`compile "org.hibernate:hibernate-validator:${hibernate_validator_version}"` is required.

For maven I removed the dependency because already imported by spring-boot-starter-web.
I just factorized the version in the properties

**Comment by gunnarmorling** [comment]
> For maven I removed the dependency because already imported by spring-boot-starter-web.

Ah, so you are overriding the property originally declared in `spring-boot-dependencies`. That should work, just let's hope the variable name doesn't get changed upstream :)

**Comment by ctamisier** [comment]
hmm I see so maybe it's not really a good idea to remove it.
And this variable being overriden in maven was because it was like this in gradle (but actually not used). The need to upgrade to HV to 5.3.3.Final wasn't wanted from what I understand in this comment https://github.com/jhipster/generator-jhipster/pull/4541#issuecomment-268892810 ?
Should we just use the one from spring-boot-dependencies ?

**Comment by gunnarmorling** [comment]
> The need to upgrade to HV to 5.3.3.Final wasn't wanted from what I understand in this comment 

It was wanted (sorry, had put my other comment to #4541 accidentally instead of this PR). HV 5.3 has many advantages over 5.2, hence I sent PR #4541 to pull 5.3 into JHipster apps.

**Comment by ctamisier** [comment]
okay.
I revert the change about the dependency (you're right, we can't rely on the variable name in the parent and override it).

**Pull request merged**